lib/views/home_content/feed/post_card/votes_widget.dart

```diff
@@ -1,43 +1,59 @@
 import "package:flutter/material.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_id_firestore.dart";
+import "package:proxima/models/database/vote/upvote_state.dart";
+import "package:proxima/viewmodels/upvote_view_model.dart";
 
 /// This widget is used to display the votes of a post.
 /// It contains two buttons to upvote and downvote the post and the number of votes.
-class VotesWidget extends StatelessWidget {
-  final int votes;
+class VotesWidget extends HookConsumerWidget {
+  final PostIdFirestore postId;
 
   const VotesWidget({
     super.key,
-    required this.votes,
+    required this.postId,
   });
 
   @override
-  Widget build(BuildContext context) {
+  Widget build(BuildContext context, WidgetRef ref) {
+    final asyncPostVote = ref.watch(postVoteProvider((postId: postId)));
+    final votes = asyncPostVote.valueOrNull?.votes ?? 0;
+    final upvoteState =
+        asyncPostVote.valueOrNull?.upvoteState ?? UpvoteState.none;
+
+    final upvoteNotifier =
```

Comment:
Please stack all `ref.` accesses to the top of the build method as they can introduce side effects and are quite tricky to track otherwise.

Reply:
Good advice !
I do that in b43e73420a4775989b68de80142e6329265da0d7.